code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,542 @@
+
+	for(var/progress = 0, progress <= 3, progress++)
+		switch(progress)
+			if(1)
+				usr << "<span class='notice'>You begin allocating energy for the enthralling.</span>"
+				usr.visible_message("<span class='danger'>[usr]'s eyes begin to throb a piercing red.</span>")
+			if(2)
+				usr << "<span class='notice'>You begin the enthralling of [target].</span>"
+				usr.visible_message("<span class='danger'>[usr] leans over [target], their eyes glowing a deep crimson, and stares into their face.</span>")
+				target << "<span class='userdanger'>Your gaze is forcibly drawn into a blinding red light. You fall to the floor as conscious thought is wiped away.</span>"
+				target.Weaken(10)
+			if(3)
+				usr << "<span class='notice'>You begin rearranging [target]'s memories.</span>"
+				usr.visible_message("<span class='danger'>[usr]'s eyes flare brightly, and a horrible grin begins to spread across [target]'s face...</span>")
+				target << "<span class='userdanger'>Your head cries out. The veil of reality begins to crumple and something evil bleeds through.</span>" //Ow the edge
+		if(!do_mob(usr, target, 100)) //around 30 seconds total for enthralling
+			usr << "<span class='danger'>The enthralling has been interrupted - your target's mind returns to its previous state.</span>"
+			target << "<span class='userdanger'>You realize you've just avoided a terrible fate. Your thoughts become coherent once more.</span>"
+			enthralling = 0
+			return
+
+	enthralling = 0
+	usr << "<span class='notice'>You have enthralled <b>[target]</b>!</span>"
+	target << "<span class='deadsay'><b>You see the Truth. Reality has been torn away and you realize what a fool you've been.</b></span>"
+	target << "<span class='deadsay'><b>The shadowlings are your masters.</b> Serve them above all else and ensure they complete their goals.</span>"
+	target << "<span class='deadsay'>You may not harm other thralls or the shadowlings. However, you do not need to obey other thralls.</span>"
+	target << "<span class='deadsay'>You can communicate with the other enlightened ones by using the Hivemind Commune ability.</span>"
+	target.adjustOxyLoss(-200) //In case the shadowling was choking them out
+	ticker.mode.add_thrall(target.mind)
+	target.mind.special_role = "Thrall"
+	hardset_dna(target, null, null, null, null, /datum/species/human/thrall)
+	var/datum/mind/thrall_mind = target.mind
+	thrall_mind.current.verbs += /mob/living/carbon/human/proc/shadowling_commune //Lets thralls hive-chat
+
+
+/mob/living/carbon/human/proc/shadowling_commune()
+	set category = "Shadowling"
+	set name = "Hivemind Commune"
+
+	var/text = stripped_input(usr, "What do you want to say to fellow thralls and shadowlings?.", "Hive Chat", "")
+	if(!text)
+		return
+	for(var/mob/M in mob_list)
+		if(is_shadow_or_thrall(M) || M in dead_mob_list)
+			M << "<span class='shadowling'><b>\[Hive Chat\]</b><i> [usr.real_name]</i>: [text]</span>"
```

Comment:
Change this line to if(is_shadow_or_thrall(M) || (M in dead_mob_list))

Basically, the "in" keyword is being done last. Instead of what you're assuming is happening, seeing if is_shadow_or_thrall(M) is true || M in dead_mob_list is true, byond is instead doing (is_shadow_or_thrall(M) || M) first and then seeing if the result of that is in dead_mob_list. null || mob returns mob, which is why dead chat worked. However, 1 is not in dead_mob_list, which is why thralls couldn't see it.

If that explaination is confusing, just take my word for it and remember to always surround x in list statements with ().